package/setup.py

```diff
@@ -36,7 +36,7 @@
     long_description_content_type="text/markdown",
     license="Apache Software License (Apache 2.0)",
     url="https://github.com/kedro-org/kedro-viz",
-    python_requires=">=3.7, <3.11",
+    python_requires=">=3.7, <3.12",
```

Comment:
For the record, Kedro and kedro-datasets are now lifting the upper bound completely.

If Kedro-Viz has a strong feeling against protecting users from installing it on a version that is too new, you might want to add a warning like we did in https://github.com/kedro-org/kedro/pull/2742

Bear in mind that Python 3.12 is out two months from now